src/docfx/build/xref/XrefMap.cs

```diff
@@ -156,20 +153,49 @@ private static XrefSpec LoadSchemaDocument(List<Error> errors, JToken token, Doc
 
                 var (schemaErrors, content) = JsonUtility.ToObject(token, schema.Type, transform: TransformContent);
                 errors.AddRange(schemaErrors);
-                return xref;
+                return xrefs.Values;
 
-                object TransformContent(DataTypeAttribute attribute, object value)
+                object TransformContent(DataTypeAttribute attribute, object value, string jsonPath, string fieldName)
                 {
                     if (attribute is UidAttribute)
                     {
-                        if (!string.IsNullOrEmpty((string)value))
+                        var level = GetLevelFromPath(jsonPath);
+                        if (!xrefs.ContainsKey(level))
+                        {
+                            xrefs.Add(level, new XrefSpec { Uid = (string)value, Href = file.SitePath });
+                        }
+                        else
+                        {
+                            errors.Add(Errors.MultipleUid());
+                        }
+                    }
+
+                    if (attribute is XrefPropertyAttribute)
+                    {
+                        if (!xrefs.TryGetValue(GetLevelFromPath(jsonPath), out var xref))
+                        {
+                            errors.Add(Errors.UidMissing());
+                        }
+                        else
                         {
-                            // TODO: set name and extension data based on defined xref properties in the schema type
-                            xref = new XrefSpec { Uid = (string)value, Href = file.SitePath };
+                            if (fieldName == "Name")
+                            {
+                                xref.Name = (string)value;
+                            }
+                            else
+                            {
+                                xref.ExtensionData[fieldName] = (string)value;
+                            }
                         }
                     }
                     return (string)value;
                 }
+
+                // e.g.: name => 1, data.name => 2
+                int GetLevelFromPath(string path)
```

Comment:
>GetLevelFromPath [](start = 20, length = 16)

how about a document with multiple uids that belong to different stuffs?
a:
  uid: x
  property: a
b: 
  uid: y
  property: b